Changelog throws 500 in SSR mode

### I tried this:

We currently try to migrate away from static builds (it takes just too much time) to the SSR mode. However we found several issues. 

One is as soon as we have an versioned message (event) and one clicks on the "Changelog" menu entry in the left sidebar the server throws a 500.


### This happened:

```
8:44:06 [ERROR] Error: ENOENT: no such file or directory, scandir '../domains/mydomain/events/MyEvent'
    at async readdir (node:internal/fs/promises:956:18)
    at async getFilesForDiffInCollection (file:///home/service/dist/server/chunks/index_xPoZ3Tqw.mjs:83:10)
    at async Promise.all (index 0)
    at async getFilesDiffsBetweenVersions (file:///home/service/dist/server/chunks/index_xPoZ3Tqw.mjs:99:53)
    at async getDiffsForCurrentAndPreviousVersion (file:///home/service/dist/server/chunks/index_xPoZ3Tqw.mjs:120:16)
```

What I can say for sure is that `/home/service/domains/mydomain/events/MyEvent` exists and is a folder with a `index.mdx`,`changelog.mdx`, a `versioned` folder and other things. Im not quite sure what the cwd is for the call above.

### I expected this:

It should render the changelog of the event. The interesting bit is that as soon as we only have one version of something it works. The difference on the file level is that the entity with only a single version has no "versioned" folder, the other ones have one.

### Is there a workaround?

Not known

### Anything else?

We basically use the setup from https://www.eventcatalog.dev/docs/development/deployment/hosting-options 

### EventCatalog Version

4.1.3

### Node.js Version

24

### Platform(s)

_No response_

### Community Notes

<!-- Please keep this note for the community -->
* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

--- a/.changeset/calm-files-resolve.md
+++ b/.changeset/calm-files-resolve.md
@@ -1,0 +1,5 @@
+---
+'@eventcatalog/core': patch
+---
+
+Fix changelog file diff resolution for versioned resources in SSR deployments.

--- a/packages/core/eventcatalog/src/utils/__tests__/collections/file-diffs.spec.ts
+++ b/packages/core/eventcatalog/src/utils/__tests__/collections/file-diffs.spec.ts
@@ -110,5 +110,37 @@ describe('Diff Utilities', () => {
       expect(diffs[0]).toContain('<span class="d2h-file-name">asyncapi.yml</span>');
       expect(diffs[1]).toContain('<span class="d2h-file-name">schema.json</span>');
     });
+
+    it('resolves Astro relative file paths from PROJECT_DIR when diffing in SSR mode', async () => {
+      const originalProjectDir = process.env.PROJECT_DIR;
+      process.env.PROJECT_DIR = pathToTestCatalog;
+
+      const allEvents = [
+        {
+          data: { id: 'myevent', version: '1.0.0' },
+          filePath: '../events/OrderAmended/index.mdx',
+        },
+        {
+          data: { id: 'myevent', version: '0.9.0' },
+          filePath: '../events/OrderAmended/versioned/0.0.1/index.mdx',
+        },
+      ];
+
+      try {
+        const diffs = (await getDiffsForCurrentAndPreviousVersion(
+          '1.0.0',
+          '0.9.0',
+          'myevent',
+          // @ts-ignore
+          allEvents
+        )) as string[];
+
+        expect(diffs).toHaveLength(2);
+        expect(diffs[0]).toContain('<span class="d2h-file-name">asyncapi.yml</span>');
+        expect(diffs[1]).toContain('<span class="d2h-file-name">schema.json</span>');
+      } finally {
+        process.env.PROJECT_DIR = originalProjectDir;
+      }
+    });
   });
 });

--- a/packages/core/eventcatalog/src/utils/file-diffs.ts
+++ b/packages/core/eventcatalog/src/utils/file-diffs.ts
@@ -1,8 +1,9 @@
 import { readdir, readFile } from 'node:fs/promises';
-import { dirname, join } from 'node:path';
+import { dirname, isAbsolute, join } from 'node:path';
 import { formatPatch, structuredPatch } from 'diff';
 import { html, parse } from 'diff2html';
 import { getItemsFromCollectionByIdAndSemverOrLatest } from './collections/util';
+import { getAbsoluteFilePathForAstroFile } from './files';
 import type { CollectionEntry } from 'astro:content';
 import type { CollectionTypes } from '@types';
 
@@ -32,7 +33,8 @@ export async function getFilesForDiffInCollection(
   const pathToFolder = collection.filePath;
   if (!pathToFolder) return [];
 
-  const dir = dirname(pathToFolder);
+  const absolutePathToFolder = isAbsolute(pathToFolder) ? pathToFolder : getAbsoluteFilePathForAstroFile(pathToFolder);
+  const dir = dirname(absolutePathToFolder);
   const allFilesInDirectory = await readdir(dir);
 
   return allFilesInDirectory

